src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/annotations/PostConstructReturnTypeQuickFix.java

```diff
@@ -33,19 +42,55 @@
  * @author Yijia Jing
  *
  */
-public class PostConstructReturnTypeQuickFix {
+public class PostConstructReturnTypeQuickFix implements IJavaCodeActionParticipant {
+
+    private static final Logger LOGGER = Logger.getLogger(PostConstructReturnTypeQuickFix.class.getName());
+
+    private final static String TITLE_MESSAGE = Messages.getMessage("ChangeReturnTypeToVoid");
+    @Override
+    public String getParticipantId() {
+        return PostConstructReturnTypeQuickFix.class.getName();
+    }
+
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic) {
         List<CodeAction> codeActions = new ArrayList<>();
-        PsiElement node = context.getCoveredNode();
-        PsiMethod parentType = getBinding(node);
-        String name = Messages.getMessage("ChangeReturnTypeToVoid");
-        ChangeCorrectionProposal proposal = new ModifyReturnTypeProposal(name, context.getSource().getCompilationUnit(),
-                context.getASTRoot(), parentType, 0, PsiPrimitiveType.VOID);
-        CodeAction codeAction = context.convertToCodeAction(proposal, diagnostic);
-        codeActions.add(codeAction);
+        final PsiElement node = context.getCoveredNode();
+        final PsiMethod parentType = getBinding(node);
+        if (parentType != null) {
+            codeActions.add(createCodeAction(context, diagnostic));
+        }
         return codeActions;
     }
 
+    @Override
+    public CodeAction resolveCodeAction(JavaCodeActionResolveContext context) {
+        final CodeAction toResolve = context.getUnresolved();
+        final PsiElement node = context.getCoveredNode();
+        final PsiMethod parentType = getBinding(node);
+        assert parentType != null;
+        ChangeCorrectionProposal proposal = new ModifyReturnTypeProposal(TITLE_MESSAGE, context.getCompilationUnit(),
+                context.getASTRoot(), parentType, 0, PsiPrimitiveType.VOID);
+        try {
+            WorkspaceEdit we = context.convertToWorkspaceEdit(proposal);
+            toResolve.setEdit(we);
+        } catch (Exception e) {
+            LOGGER.log(Level.WARNING, "Unable to create workspace edit for code action to change return type to void", e);
+        }
+        return toResolve;
+    }
+
+    private CodeAction createCodeAction(JavaCodeActionContext context, Diagnostic diagnostic) {
+        ExtendedCodeAction codeAction = new ExtendedCodeAction(TITLE_MESSAGE);
+        codeAction.setRelevance(0);
+        codeAction.setDiagnostics(Collections.singletonList(diagnostic));
+        codeAction.setKind(CodeActionKind.QuickFix);
+        codeAction.setData(new CodeActionResolveData(context.getUri(), getParticipantId(),
+                context.getParams().getRange(), Collections.emptyMap(),
+                context.getParams().isResourceOperationSupported(),
+                context.getParams().isCommandConfigurationUpdateSupported()));
+        return codeAction;
+    }
```

Comment:
It seems that this method contains common code that many of the quick fixes would use. We should be reusing the code instead of duplicating it in each quick fix.

Reply:
@mrglavas - can we raise a separate PR for moving this method to a common file?

Reply:
@vaisakhkannan Yes. Feel free to open a new GitHub issue for that change. Thanks.